### PR TITLE
arch: move help strings to a markdown file

### PR DESCRIPTION
**Pull request by zleyyij**
This change moves the `about` and `summary` help strings to `arch.md` in the `arch` directory

### CODE REVIEW

**Comment by zleyyij** [comment]
#4368 
mentioning the relevant issue

File: src/uu/arch/arch.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+```
+arch
+```
+
+## About
+```
+Display machine architecture
+```
+
+## Summary
```
Comment (tertsdiepraam) [inline src/uu/arch/arch.md:14]: Could you use the name `After help` here for consistency with other utils?

File: src/uu/arch/src/arch.rs [hunk 2]
```diff
@@ -11,8 +11,8 @@ use platform_info::*;
 use clap::{crate_version, Command};
 use uucore::error::{FromIo, UResult};
 
-static ABOUT: &str = "Display machine architecture";
-static SUMMARY: &str = "Determine architecture name for current machine.";
+static ABOUT: &str = help_section!("about", "arch.md");
```
Comment (tertsdiepraam) [inline src/uu/arch/src/arch.rs:14]: Some errors here (you need to import the macros)

File: src/uu/numfmt/numfmt.md [hunk 3]
```diff
@@ -1,4 +1,3 @@
-<!-- spell-checker:ignore N'th M'th -->
```
Comment (tertsdiepraam) [inline src/uu/numfmt/numfmt.md:1]: Why did you delete this? Now the spell checker is complaining :)

File: src/uu/arch/arch.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<!-- spell-checker:ignore N'th M'th -->
```
Comment (tertsdiepraam) [inline src/uu/arch/arch.md:1]: This line is not necessary here because those words don't appear in the file.

**Review by tertsdiepraam — COMMENTED** [review]
Nice! Thanks! I have a few small suggestions.

File: src/uu/numfmt/numfmt.md (hunk 3 above)
Comment (zleyyij) [reply src/uu/numfmt/numfmt.md:1]: I deleted it in the wrong file, so sorry about that

**Comment by zleyyij** [comment]
All those changes have been applied, thank you for the feedback :)

File: src/uu/arch/src/arch.rs [hunk 5]
```diff
@@ -10,9 +10,10 @@ use platform_info::*;
 
 use clap::{crate_version, Command};
 use uucore::error::{FromIo, UResult};
+use uucore::{help_section};
```
Comment (sylvestre) [inline src/uu/arch/src/arch.rs:13]: please run rustfmt :)

File: src/uu/arch/arch.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# arch
+
+```
+arch
+```
+
+```
```
Comment (tertsdiepraam) [inline src/uu/arch/arch.md:7]: The code fences for about and after help will lead to weird formatting. (The first fences for the usage are correct though)

File: src/uu/numfmt/numfmt.md (hunk 3 above)
Comment (tertsdiepraam) [reply src/uu/numfmt/numfmt.md:1]: You haven't undone the change yet :)

File: src/uu/arch/src/arch.rs [hunk 7]
```diff
@@ -10,9 +10,10 @@ use platform_info::*;
...
 
-static ABOUT: &str = "Display machine architecture";
-static SUMMARY: &str = "Determine architecture name for current machine.";
+static ABOUT: &str = help_section!("about", "arch.md");
```
Comment (tertsdiepraam) [inline src/uu/arch/src/arch.rs:15]: See https://github.com/uutils/coreutils/pull/4385 for why this is failing

**Comment by zleyyij** [comment]
Those changes should be applied, I remembered to use rustfmt :)

**Comment by tertsdiepraam** [comment]
Thanks!
